src/legacy.erl

```diff
@@ -0,0 +1,286 @@
+-module legacy.
+
+%%% GENERAL CONSIDERATIONS
+% > In dynamically typed languages, where we don't really have classes, it's not enough to fake
+%   the parameters to functions in modules or build fake objects for things returned by certain
+%   functions (i.e. when you have V = this:thing(...), it's not enough to be able to mock 'this'
+%   or 'thing' or provide fake parameters to it, you also need to figure out what 'this:thing'
+%   may evaluate to, which is not trivially determined by "the type of V" since V has no type).
+
+
+%%% ADAPT PARAMETER
+-spec adapt_parameter(http_servlet_request:t()) -> _.
+adapt_parameter(HttpRequest) ->
+    Values = http_servlet_request:parameter_values(HttpRequest),
+    do:some_stuff(with, Values).
+
+somewhere_else(HttpRequest) ->
+    adapt_parameter(HttpRequest).
+
+-spec new_adapt_parameter(parameter_source:t()) -> _.
+new_adapt_parameter(ParamSource) ->
+    Values = parameter_source:values(HttpRequest),
+    do:some_stuff(with, Values).
+
+new_somewhere_else(HttpRequest) ->
+    % http_request_param_source implements behavior parameter_source
+    new_adapt_parameter(http_request_param_source:new(HttpRequest)).
+
+
+%%% BREAK OUT METHOD OBJECT
```

Comment:
I get it more like

    we have a gen_server,  we need to modify a given handle_call (to say something).  It isn't tested and it isn't  easy 
    to test because it is difficult to get such a process running in a test case (depends on other process to be alive,  
    do something awful on init(),  etc).    So:   move  the handle_call into its own function.    Pass only the arguments 
    really needed.   Then test that function,  that doesn't require the gen_server to be spawn (and _hopeful_ has less 
    dependencies or those could be resolved easier than the ones for the full process)